corehq/apps/locations/tests/test_change_administrative_status.py

```diff
@@ -0,0 +1,133 @@
+"""
+When a location type is set from stock-tracking to not stock-tracking, find all
+locations of that type and:
+    close the supply point case,
+    nullify the supply_point_id,
+    nullify the StockState sql_location field
+
+When a location type is set from not stock tracking to stock tracking, find all
+locations of that type and:
+see if there is a closed supply point case with that location id
+if so:
+    reopen that case
+    set the supply_point_id to that
+    set the sql_location field on any appropriate stock states
+otherwise:
+    open a new supply point case as normal
+"""
+import datetime
+
+from casexml.apps.stock.const import SECTION_TYPE_STOCK
+from django.test import TestCase
+
+from corehq.apps.commtrack.models import SupplyPointCase, StockState
+from corehq.apps.commtrack.tests.util import bootstrap_domain
+from corehq.apps.products.models import SQLProduct
+
+from ..models import SQLLocation
+from .util import setup_locations_and_types
+
+
+class TestChangeStatus(TestCase):
+    domain = 'test-change-administrative'
+    location_type_names = ['state', 'county', 'city']
+    stock_tracking_types = ['city']
+    location_structure = [
+        ('Massachusetts', [
+            ('Middlesex', [
+                ('Cambridge', []),
+                ('Somerville', []),
+            ]),
+            ('Suffolk', [
+                ('Boston', []),
+            ])
+        ])
+    ]
+
+    def setUp(self):
+        self.domain_obj = bootstrap_domain(self.domain)
+
+        self.location_types, self.locations = setup_locations_and_types(
+            self.domain,
+            self.location_type_names,
+            self.stock_tracking_types,
+            self.location_structure,
+        )
+
+        self.suffolk = self.locations['Suffolk']
+        self.boston = self.locations['Boston']
+        self.county_type = self.location_types['county']
+        self.city_type = self.location_types['city']
+
+    def tearDown(self):
+        self.domain_obj.delete()
+
+    def assertHasSupplyPoint(self, location):
+        msg = "'{}' does not have a supply point.".format(location.name)
+        loc = SQLLocation.objects.get(location_id=location.location_id)
+        self.assertIsNotNone(loc.supply_point_id, msg)
+        self.assertIsNotNone(loc.linked_supply_point(), msg)
+
+    def assertHasNoSupplyPoint(self, location):
+        msg = "'{}' should not have a supply point".format(location.name)
+        loc = SQLLocation.objects.get(location_id=location.location_id)
+        self.assertIsNone(loc.supply_point_id, msg)
+
+    def test_change_to_track_stock(self):
+        self.assertHasSupplyPoint(self.boston)
```

Comment:
I was thinking of this as trying to illustrate what's happening - "first it has a supply point, then you set it to administrative, then it doesn't have a supply point, then you set it back..."

It's definitely not necessary to have in each one, but I think it makes the intent more obvious.